src/main/ruby/truffleruby/core/match_data.rb

```diff
@@ -153,6 +163,21 @@ def match_length(n)
   def to_s
     self[0]
   end
+
+  private
+
+  def backref_from_arg(index)
+    if Primitive.is_a?(index, String) || Primitive.is_a?(index, Symbol)
+      names_to_backref = Hash[Primitive.regexp_names(self.regexp)]
```

Comment:
Creating a Hash to just lookup once/twice in it seems expensive, but it was already like that.
Probably we should add a primitive taking a name and returning the backrefs or last backref.
But the code was already like that, so let's keep as-is in this PR.